cli/src/migrations/pgroll.ts

```diff
@@ -29,3 +30,86 @@ export async function allMigrationsPgRollFormat() {
   }
   return true;
 }
+
+const getPgRollLink = (table: any, column: any) => {
+  const foreignKeysForTable = table.foreignKeys;
+  const foreignKeys = Object.keys(foreignKeysForTable ?? {});
+  for (const key of foreignKeys) {
+    const k = foreignKeysForTable[key];
+    if (k && k.columns?.includes(column.name)) {
+      return k;
+    }
+  }
+  return null;
+};
+
+function pgRollToXataColumnType(type: string): string {
+  switch (type) {
+    case 'boolean':
+      return 'bool';
+    case 'bigint':
+    case 'integer':
+      return 'int';
+    case 'double precision':
+      return 'float';
+    case 'text':
+      return 'text';
+    case 'timestamptz':
+      return 'datetime';
+    case 'text[]':
+      return 'multiple';
+    case 'jsonb':
+      return 'json';
+    case 'xata_file':
+      return 'file';
+    case 'xata_file_array':
+      return 'file[]';
+    case 'real[]':
+      return 'vector';
+    default:
+      return type;
+  }
+}
+
+export async function getBranchDetailsWithPgRoll(
```

Comment:
By inverting the logic do you mean that this function will be exported in client? Or just the `pgRollToXataColumnType` mapping maybe?

Reply:
Ideally none, let's see in future PRs how we handle it